onchain/rollups/test/foundry/consensus/authority/AuthorityFactory.t.sol

```diff
@@ -41,41 +42,63 @@ contract AuthorityFactoryTest is Test {
     function testNewAuthority(address _authorityOwner) public {
         vm.assume(_authorityOwner != address(0));
 
-        // expect event emitted from the authority contract
-        vm.expectEmit(false, false, false, true);
-        emit ConsensusCreated(_authorityOwner, inputBox);
```

Comment:
`vm.expectEmit` is a quick and simple way to test the emittance of `ConsensusCreated`. I don't think it's necessary to use `vm.recordLogs();` for this event. But I'm not against it either.

Reply:
The `vm.expectEmit` is limited as it doesn't check whether two events were emitted.

Reply:
But I'm curious to know why you think it is unnecessary to use `vm.recordLogs` for checking the emission of `ConsensusCreated`, but necessary for `AuthorityCreated`.

Reply:
Because the event contains the deployed address of authority, which we only know after the deployment. So we didn't know what to "expect" beforehand.
I tried to explain this in the following 2 lines of comments. But if we were to use `vm.recordLogs()` for all events, then indeed those 2 lines should be removed.

Reply:
Oh, I see now! Yeah, that makes sense now. Can I leave the usage of `vm.recordLogs` for `ConsensusCreated` just because it is more powerful than `vm.expectEmit` in detecting multiple event emission?

Reply:
Yes. We can keep it for sure :)